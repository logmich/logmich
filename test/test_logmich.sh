#!/bin/sh

# Usage: test_logmich.sh EXENAME OUTPUTPREFIX ABSSOURCEDIR

set -e

TEST_STDOUT="$(mktemp)"
TEST_STDERR="$(mktemp)"

echo "Running $1..."

"$1" 2> "${TEST_STDERR}" > "${TEST_STDOUT}"

# strip the absolute path in the test output
sed -i "s#${3}/##" "${TEST_STDERR}" "${TEST_STDOUT}"

if cmp -s "${TEST_STDOUT}" "${2}.stdout"; then
  echo "stdout ok"
else
  echo "stdout mismtach:"
  diff -u "${2}.stdout" "${TEST_STDOUT}"
  exit 1
fi

if cmp -s "${TEST_STDERR}" "${2}.stderr"; then
  echo "stderr ok"
else
  echo "stderr mismtach:"
  diff -u "${2}.stderr" "${TEST_STDERR}"
  exit 1
fi

echo "$1 test finished successfully"

exit 0

# EOF #

