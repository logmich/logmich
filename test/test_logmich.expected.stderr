[INFO ] test/test_logmich.cpp:38: -
[WARN ] test/test_logmich.cpp:39: -
[ERROR] test/test_logmich.cpp:40: -
[FATAL] test/test_logmich.cpp:44: unreachable code reached
[ERROR] test/test_logmich.cpp:45: not implemented
[ERROR] test/test_logmich.cpp:48: error level log message
[WARN ] test/test_logmich.cpp:49: warring level log message
[INFO ] test/test_logmich.cpp:50: info level log message
[ERROR] test/test_logmich.cpp:55: error level log message with format: 5
[WARN ] test/test_logmich.cpp:56: warring level log message with format: 10
[INFO ] test/test_logmich.cpp:57: info level log message with format: Hello World
[DEBUG] test/test_logmich.cpp:58: debug level log message with format: 1 2 3 4
[TRACE] test/test_logmich.cpp:59: tmp level log message 42
[DEBUG] test/test_logmich.cpp:64: this should be visible
[TRACE] test/test_logmich.cpp:65: this should be visible
[WARN ] test/test_logmich.cpp:67: format test 1
[WARN ] test/test_logmich.cpp:68: format test 1 2 1
[WARN ] test/test_logmich.cpp:69: format test 1
[WARN ] test/test_logmich.cpp:70: format test 1
[WARN ] test/test_logmich.cpp:71: format test <custom-type>
